src/instructlab/sdg/utils/parse_and_convert.py

```diff
@@ -0,0 +1,313 @@
+        raise utils.GenerateException(
+            "Generated message cannot be converted to legacy format"
+        )
+    sample["system"] = _unescape(sample["messages"][0]["content"])
+    sample["user"] = _unescape(sample["messages"][1]["content"])
+    sample["assistant"] = _unescape(sample["messages"][2]["content"])
+
+    return sample
+
+
+def _convert_to_hack_fmt(sample: dict, sys_prompt: str):
+    """
+    Convert a sample dictionary to contain 'system', 'user', and 'assistant' columns.
+
+    Note: We should remove this function in the future when we resolve this issue and
+    standardize the format to messages.
+    """
+    # Create user query message
+    user_query = _unescape(_get_question(sample))
+    response = _unescape(_get_response(sample))
+    if "context" in sample:
+        user_query = f"{sample['context']}\n\n{user_query}"
+
+    sample["id"] = str(uuid.uuid4())
+    sample["system"] = sys_prompt
+    sample["user"] = user_query
+    sample["assistant"] = response
+
+    return sample
+
+
+def _convert_to_messages(sample: dict, sys_prompt: str):
+    """
+    Convert a sample dictionary to contain 'messages'
+    and 'metadata' columns required for training.
+    """
+    # Create user query message
+    user_query = _unescape(_get_question(sample))
+    response = _unescape(_get_response(sample))
+
+    sample["id"] = str(uuid.uuid4())
+    sample["messages"] = [
+        {"content": sys_prompt, "role": "system"},
+        {"content": user_query, "role": "user"},
+        {"content": response, "role": "assistant"},
+    ]
+
+    return sample
+
+
+def create_auxiliary_dataset(generated_dataset: Dataset):
+    if "dataset_type" not in generated_dataset.column_names:
+        return None
+    auxilary_path = "src/instructlab/sdg/config/kowledge/auxilary_instructions.yaml"
+    if os.path.isfile(auxilary_path):
+        with open(auxilary_path, "r", encoding="utf-8") as fp:
+            auxiliary_inst = yaml.safe_load(fp)
+    else:
+        return None
+    auxiliary_ds = generated_dataset.filter(
+        lambda x: x["dataset_type"] != "base_document"
+    )
+    unique_document_auxiliary = auxiliary_ds.to_pandas().drop_duplicates(
+        subset=["document"]
+    )
+    unique_document_auxiliary = Dataset.from_pandas(
+        unique_document_auxiliary
+    ).remove_columns(
+        [
+            col
+            for col in unique_document_auxiliary.column_names
+            if col
+            not in [
+                "raw_document",
+                "document_outline",
+                "domain",
+                "dataset_type",
+                "document",
+            ]
+        ]
+    )
+    print(unique_document_auxiliary)
+    unique_document_auxiliary = unique_document_auxiliary.rename_columns(
+        {"raw_document": "context", "document": "response"}
+    )
+
+    def __create_auxiliary_ds(rec):
+        instruction = random.choice(auxiliary_inst[rec["dataset_type"]])
```

Comment:
Unclear to me where `dataset_type` is coming from and what it's valid values are?